### ISSUE TITLE
Callout A11y Pass

### ISSUE THREAD

**Issue by awalia13** [OfficeDev/office-ui-fabric-react#8616]
Tracking all of the issues brought up with the 04/02 accessibility pass on Callout control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B6ca2ab89-695a-49ad-a4b8-3eae316c2987%7D&action=default&uid=%7B6CA2AB89-695A-49AD-A4B8-3EAE316C2987%7D&ListItemId=1922&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)
 - [x] Bug1:[Accessibility][Programmatic access]: Name and title attribute is not defined for more(…) button.
 - [x] Bug2:[Accessibility][Keyboard Navigation]: Focus moves out of the opened callout upon pressing TAB.
 - [ ] Bug3:[Accessibility][Keyboard Navigation]: Focus moves to the top of the page upon closing the opened callout using ESC.
 - [x] Bug4:[Accessibility][Keyboard Navigation]: Callout elements under Callout with cover and Callout with directional hint  variant is not at all accessible by keyboard.
 - [x] Bug5:[Accessibility][Usable]: In Scan mode of narrator focus moves to the new and upload icon but it does not narrate anything.
 - [x] Bug6:[Accessibility][Supporting the Platform]: Go to Microsoft link is not adapting the suggested color in High contrast modes(Black and white).
 - [ ] Bug7:[Accessibility][Screen Reader]: No information about the role is provided about the opened dialog when open callout button is clicked
 - [ ] Bug8:[Accessibility][Usable]: Screen does not scroll down along with keyboard focus in Backward navigation.

**Comment by micahgodbolt** [comment]
#1 i can add the name, but it seems that title is redundant and unnecessary on buttons.

**Comment by micahgodbolt** [comment]
# 2 - going to remove that example. To fix issue we'd add a focus trap which is the same as the example next to it

**Comment by micahgodbolt** [comment]
4- onKeyPress (to close via escape) was only getting called when focus was on the callout. I added a window level listener for escape to call onKeyPress as well.

**Comment by micahgodbolt** [comment]
5 is A11yUsable. will come back to that later.

**Comment by micahgodbolt** [comment]
6- appears to be fixed now. 

![image](https://user-images.githubusercontent.com/1434956/65113673-f0387900-d998-11e9-8be3-febfd220a2f5.png)

![image](https://user-images.githubusercontent.com/1434956/65113680-fa5a7780-d998-11e9-8fad-8d1d78d3bbf4.png)

**Comment by micahgodbolt** [comment]
7 - Still trying to understand the criteria for narrator to announce the role of a callout. 

This example will say "Alert, then read the title of the callout"

http://www.oaa-accessibility.org/examplep/alertdialog1/

Our example is the same, but it doesn't announce the role. Might need to followup with Narrator team to understand why.

**Comment by micahgodbolt** [comment]
Bug 3 tracked in #10574

**Comment by micahgodbolt** [comment]
Bug 3 tracked in #10485

**Comment by micahgodbolt** [comment]
All issues moved out to their own issue or resolved. Closing this on down.

**Comment by 999-Dev** [comment]
> 1- i can add the name, but it seems that title is redundant and unnecessary on buttons.

@micahgodbolt  
We verify those bugs in below Test Enviourment.

Test Environment: 
OS: Version: RS5_Prerelease(1903 OS Build 18990.1) 
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/callout

Agreed, Screen reader users will not be able to know about the button, if name is not present. So just add the name of the control.

**Comment by 999-Dev** [comment]
Bug5:[Accessibility][Usable]: In Scan mode of narrator focus moves to the new and upload icon but it does not narrate anything.
We verify those bugs in below Test Enviourment and found issue is fixed.
Test Environment: 
OS: Version: RS5_Prerelease(1903 OS Build 18990.1) 
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/callout
Edge Browser Version: 
Microsoft Edge: 44.18990.1.0
Microsoft EdgeHTML: 18.18990
Chrome Browser Version:
Chrome Version: 76.0.3809.132
 
Bug6:[Accessibility][Supporting the Platform]: Go to Microsoft link is not adapting the suggested color in High contrast modes(Black and white).
We verify those bugs in below Test Enviourment and found issue is fixed.
Test Environment: 
OS: Version: RS5_Prerelease(1903 OS Build 18990.1) 
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/callout
Edge Browser Version: 
Microsoft Edge: 44.18990.1.0
Microsoft EdgeHTML: 18.18990
Chrome Browser Version:
Chrome Version: 76.0.3809.132
